### PR TITLE
Pass2 instruction translation

### PR DESCRIPTION
**Pull request by shakram02**
Pass 2 preparation
 - Added the Instruction Set module
 - Created the OneOrMore type, to reduce code redundancy
 - Starting the object code generator that converts a whole instruction
   to machine code

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,3 +20,5 @@ path = "src/bin/main.rs"
 getopts = "*"
 regex = "*"
 log = "*"
+lazy_static = "*"
+bitflags = "*"
```
Comment (oddcoder) [inline Cargo.toml]: what are those for

File: src/lib/lib.rs [hunk 2]
```diff
@@ -1,6 +1,12 @@
 /**
  * Higher modules will be declared here
  */
+#[macro_use]
+extern crate lazy_static;
+
+#[macro_use]
+extern crate bitflags;
+
```
Comment (oddcoder) [inline src/lib/lib.rs]: what are they used for ?

**Review by oddcoder — CHANGES_REQUESTED** [review]
Can you please add couple of instructions .. .say `STA`, `J` and any other 2 of your choice ?

File: Cargo.toml (hunk 1 above)
Comment (shakram02) [reply Cargo.toml]: a quick Google search will tell you, one of them is redundant

**Review by oddcoder — APPROVED** [review]

**Pull request merged**